application/src/main/java/org/togetherjava/tjbot/commands/moderation/ModerationUtils.java

```diff
@@ -376,33 +377,37 @@ record TemporaryData(Instant expiresAt, String duration) {
+     * @param guild to retrieve the guild.
+     * @param reason for the action.
      * @return the appropriate advice.
      */
-    public static String getDmAdvice(ModerationAction action, String... message) {
-        if (message.length == 3) {
+    public static String getDmAdvice(ModerationAction action,
+            @Nullable ModerationUtils.TemporaryData temporaryData, Guild guild, String reason) {
+        if (action == ModerationAction.MUTE) {
+            String durationMessage =
+                    temporaryData == null ? "permanently" : "for " + temporaryData.duration();
             return """
-                    Hey there, sorry to tell you but unfortunately you have been %s %s from the server %s.
+                    Hey there, sorry to tell you but unfortunately you have been %s %s in the server %s.
                     To get in touch with a moderator, you can simply use the **/modmail** command here in this chat. Your message will then be forwarded and a moderator will get back to you soon :thumbsup:
                     The reason for being %s is: %s
                     """
-                .formatted(action.getVerb(), message[0], message[1], action.getVerb(), message[2]);
-        } else if (ModerationAction.UNMUTE.getVerb().equals(action.getVerb())
-                || ModerationAction.UNQUARANTINE.getVerb().equals(action.getVerb())) {
+                .formatted(action.getVerb(), durationMessage, guild.getName(), action.getVerb(),
+                        reason);
+        } else if (action == ModerationAction.UNMUTE || action == ModerationAction.UNQUARANTINE) {
             return """
                     Hey there, you have been %s in the server %s.
                     This means you can now interact with others in the server again.
                     The reason for being %s is: %s
-                    """.formatted(action.getVerb(), message[0], action.getVerb(), message[1]);
+                    """.formatted(action.getVerb(), guild.getName(), action.getVerb(), reason);
         }
         return """
-                Hey there, sorry to tell you but unfortunately you have been %s, which came from the server %s.
-                To get in touch with a moderator, you can simply use the **/modmail** command here in this chat. Your message will then be forwarded and a moderator will get back to you soon :thumbsup:
+                Hey there, sorry to tell you but unfortunately you have been %s, in the server %s.
+                To get in touch with a moderator, you can simply use the %s command here in this chat. Your message will then be forwarded and a moderator will get back to you soon :thumbsup:
                 The reason for being %s is: %s
                 """
-            .formatted(action.getVerb(), message[0], action.getVerb(), message[1]);
+            .formatted(action.getVerb(), guild.getName(),
+                    MessageUtils.mentionSlashCommand(guild, "modmail"), action.getVerb(), reason);
```

Comment:
feels like u didnt test this. the mention doesnt return u a string, it returns a restaction. u cant just put that into the string. u have to do a rest action map pipeline and change the whole method to `RestAction<String>` instead of just `String`.